SalesforceCommerceCloud/pwa-kit
# Always use the SLAS `usid` for `cookie-id` recommendation requests.

# Description

When making requests to Einstein we were sometimes not sending all the required information. Specifically there were occasions where the `cookie-id` value was never sent. This would happen for new users on the site and we explicitly set the `cookie-id` value to an empty string if we didn't have a value to send.

In this PR we make 2 changes. First we ensure we don't make requests to Einstein until the customer is initialized (aka has a valid usid) and secondly as an additional defence we only set a `cookie-id` when we have have the correct value to populate that property and send it.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Wiat for customer initialization before requesting recommendations
- Refactor the Einstein client a little and have a guard on sending empty cookie values.

# How to Test-Drive This PR

- Build and run this branch.
- Clear our any application data for local host
- View the homepage and notice any network requests filtered on `einstein` should have query data for the `cookie-id`. (This value would be blank on production)
- You can confirm that this value is still being sent for registered users by logging in and refreshing the home page and looking at the same network logs.
